src/Avalonia.Controls/Automation/Peers/TextBoxAutomationPeer.cs

```diff
@@ -19,5 +29,10 @@ protected override AutomationControlType GetAutomationControlTypeCore()
         {
             return AutomationControlType.Edit;
         }
+
+        protected override string? GetHelpTextCore()
+        {
+            return Owner.Text;
```

Comment:
I haven't checked android specific code yet.
But this doesn't look right. `AutomationProperties.HelpText` is typically responsible for supportive text info, not control value.
ToolTip info is [what we return by default](https://github.com/AvaloniaUI/Avalonia/blob/8ae698aa89880f81285c42c8ad099e688868fe63/src/Avalonia.Controls/Automation/Peers/ControlAutomationPeer.cs#L123-L126). 

In case of TextBox, HelpText can return Owner.Watermark when base implementation is null or empty, I suppose. cc @grokys

Reply:
I 100% agree, however I do think this necessitates expanding the IValueProvider API to expose an event for text changes. This is because TalkBack users expect editable text fields to announce changes to the field in real-time. Either that or add a new AutomationProperty for a similar purpose. What do you think?

Reply:
There is a PropertyChanged event on AutomationPeer, and other backends already listen for it.
But you are right on that TextBox doesn't trigger this event on changes (or I am missing something).
For example, Range controls do trigger it: 
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Automation/Peers/RangeBaseAutomationPeer.cs#L27
But TextBoxAutomationPeer doesn't.

Reply:
Relevant https://github.com/AvaloniaUI/Avalonia/issues/14253

Reply:
For consistency, let's make TextBoxAutomationPeer do the same as RangeBaseAutomationPeer. Specifically, protected virtual OwnerPropertyChanged method, and raise ValueProperty property changed inside. Without HelpText changes.

Reply:
Thanks! Will do!

Reply:
All the TextChanged/TextChanging events and the Text property itself being a fully styled property were late stage changes for 11.0. So I'm not surprised the TextBoxAutomationPeer doesn't have this functionality either. It was definately an oversight.